api/middleware/tracer.go

```diff
@@ -62,6 +65,13 @@ func Tracer(tracer opentracing.Tracer) macaron.Handler {
 		// call next handler. This will return after all handlers
 		// have completed and the request has been sent.
 		macCtx.Next()
+
+		// if tracing has been disabled we return directly without calling
+		// span.Finish()
+		if noTrace, ok := macCtx.Data["noTrace"]; ok && noTrace.(bool) {
+			return
+		}
+
```

Comment:
why do all the work up to here if we don't need tracing? can't we do this at the start of this function?

Reply:
in the beginning of this function `macCtx.Data["noTrace"]` would not have been set yet, because the tracer is at the beginning of the list of handlers before the one that sets `noTrace = true`. I think it probably makes sense to leave the tracer at it's position in the middleware chain, so that we have it available in the other middlewares if we want to use it there.